scripts/darwin/metrics.sh

```diff
@@ -30,4 +47,99 @@ else
+#    appropriate files in the wiki to make the comparison.
+#
+# 3. Then the metrics app will generate figures displaying performance data (regressions should be easy to spot).
+#
+# 4. Next a wiki page file will be generated in the wiki with links to the figures. This
+#    wiki page file will then be uploaded to the parthenon wiki.
+#
+# 5. The figures will be uploaded to an empty figures branch (orphan branch) in the parthenon repository:
+#    https://github.com/lanl/parthenon/tree/figures
+#
+#    It may be necessary in the future to squash the commit history in that branch occassionally.
+#
+# 6. Finally, a status report will be sent to github and associated with commit of the current branch,
+#    this status should show up next to other status reports such as Codacy, Deepcode and gitlab ci.
+#    The status will contain a link to the wiki page where the performance metrics are posted.
+#
+# NOTE a. Why are we uploading figures to their own branch in the parthenon repository and not
+#      simply storing them in parthenon.wiki?
+#
+#      Binary files will bloat the wiki repository
+#
+#      b. Why not simply place the image files in a branch on wiki?
+#
+#      I could not figure out how to link to the images that are located in a branch
+#      in the wiki repository. from the wiki page.
+#
+#      c. Why not store them in develop in the partheon repo?
+#
+#      Again we don't want to bloat the partheon repo with binary files.
+#
+# ********** Flags and thier meanings **********
+#
+# -p path_to_matrics_application_permissions_file
+#
+# The path to the .pem file so the metrics application can authenticate with the github server.
+#
+# --analyze path_to_tst_results
+#
+# In order to analyze, the results of the tests of the local repository the metrics application needs to
+# know where performance test output is located. So for instance if you build parthenon with the command:
+#
+# cmake -S. -B build
+# cmake --build build
+#
+# and then run the tests
+#
+# cmake --build build --target tests
+#
+# Then the tst output will appear in /build/tst/regression/outputs this is what is being
+# passed to the --analyze flag
+#
+# --create
+#
+# This flag is simply to allow the metrics app to create branches on the wiki or parthenon repository if
+# needed. So for instance if the figures branch in parthenon repository was deleted for some reason
+# the metrics app would be able to recreate it.
+#
+# --post-analyze-status
+#
+# This flag indicates that the metrics app will post a status to the github repository once it is done analyzing.
+# If this flag was not set you could still post a status but it would require another call to the metrics app.
+#
+# --branch current_branch_name
+#
+# This is the branch that the local performance tests were run for.
+#
+# --target_branch target_branch_name
+#
+# This is the branch that is being merged into if it is a pull request.
+#
+# --generate-figures-on-analysis
+#
+# Does what it says.
```

Comment:
When would we not want to run with this flag?

Also, maybe change to "Generates figures during the analysis phase" or something like that.

Reply:
Good question this can probably just be set as a default. I don't want to change that in this pr as it is just documentation.